src/commands/mod.rs

```diff
@@ -2298,6 +2376,294 @@ fn cmd_new(ui: &mut Ui, command: &CommandHelper, args: &NewArgs) -> Result<(), C
     Ok(())
 }
 
+fn cmd_next(ui: &mut Ui, command: &CommandHelper, args: &NextArgs) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let edit = args.edit;
+    let amount = args.amount;
+    assert!(amount >= 1);
+    let current_wc_id = workspace_command
+        .get_wc_commit_id()
+        .ok_or_else(|| user_error("This command requires a working copy"))?;
+    let current_wc = workspace_command.repo().store().get_commit(current_wc_id)?;
+    // TODO(#NNN): This currently depends on order in which the parents are
+    // returned, make it configurable.
+    let parents = current_wc.parents();
+    let parent = parents
+        .first()
+        .ok_or_else(|| user_error("unable to determine parent"))?;
+    let parent_commit = RevsetExpression::commit(parent.id().clone());
+    // Collect all descendants of our parent.
+    // TODO(#NNN) We currently cannot deal with multiple children, which result
+    // from branches. Fix it when --interactive is implemented.
+    let mut children_revset = RevsetExpression::children(&parent_commit);
+    // let children: Vec<Commit> = RevsetExpression::descendants(&parent_commit)
+    //     .resolve(workspace_command.repo().as_ref())?
+    //     .evaluate(workspace_command.repo().as_ref())?
+    //     .iter()
+    //     .commits(workspace_command.repo().store())
+    //     .try_collect()?;
+
+    let current_id = current_wc.id();
+    let current_short = short_commit_hash(current_id);
+    // Handle the simple `jj next` call.
+    if amount == 1 {
```

Comment:
I would have hoped we didn't need a special case for `amount == 1`. I would expect something like this:

1. Find starting commit:
```
let start_id = if edit {
  wc_commit.id()
} else {
 match wc_commit.parent_ids() {
    [parent_id] => parent_id,
    _ => {
      return user_error("Cannot run `jj next` on a merge commit");
    }
  }
};
```

2. Find target commit:
```
let target_expression = RevsetExpression::commit(start_id.clone())
    .descendants_at(amount)
    .minus(RevsetExpression::commit(wc_commit.id().clone());
let target_revset = target_expression.resolve(repo)?.evaluate(repo)?;
let target = match target_revset.iter().commits().take(2).collect_vec() {
  [target] => target,
  [] => {
    return user_error("No target commit");
  }
  _ => {
    // TODO: Prompt the user instead
    return user_error("Ambiguous target commit");
  }
};
```

3. Update working-copy commit:
```
if edit {
  tx.edit(&target);
} else{
  tx.check_out(&target);
}
```

Reply:
I agree that this is cleaner, but the initial approach was the path of least resistance.

Reply:
Yes, understood. To make sure we're on the same page, do you intend to clean it up before this is ready for review? I don't think my proposal is any more code than the current version, and it should handle more cases.

Reply:
👍 , I think I dropped a whole bunch of cases while implementing the simple `next` command. Currently it only cares about actual children, while we specified the branch moving `next`.

Reply:
It's now refactored, PTAL.